Overview page for About dbt projects (#2190)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/projects.md

id: "projects"
---

A dbt project informs dbt the context of your project and how to transform your data (build your data sets). By design, dbt enforces the top-level structure of a dbt project such as the `dbt_project.yml` file, the `models` directory, the `snapshots` directory, and so on. Within the directories of the top-level, you can organize your project in any way that meets the needs of your organization and data pipeline.

> At a minimum, all a project needs is the `dbt_project.yml` project configuration file. dbt supports a number of different resources so a project may also include them such as:

Reviewer: ```suggestion
At a minimum, all a project needs is the `dbt_project.yml` project configuration file. dbt supports a number of different resources, so a project may also include:
```

| [macros](/docs/build/jinja-macros) | Blocks of code that you can reuse multiple times. |
| [docs](/docs/collaborate/documentation) | Docs for your project that you can build. |
| [sources](/docs/build/sources) | A way to name and describe the data loaded into your warehouse by your Extract and Load tools. |
| [exposures](/docs/build/exposures) | A way to define and describe a downstream use of your project. |
| [metrics](/docs/build/metrics) | A way for you to define metrics for your project. |
| [analysis](/docs/build/analyses) | A way to organize analytical SQL queries in your project such as the general ledger from your QuickBooks. |

When building out the structure of your project, you should consider these impacts to your organization's workflow:

* **How would people run dbt commands** &mdash; Selecting a path
* **How would people navigate within the project** &mdash; Whether as developers in the IDE or stakeholders from the docs
* **How would people configure the models** &mdash; Some bulk configurations are easier done at the directory level so people don’t have to remember to do everything in a config block with each new model

## Project configuration

> Every dbt project includes a project configuration file, called `dbt_project.yml`. It defines the directory of the dbt project and other project configurations.

Reviewer: ```suggestion
Every dbt project includes a project configuration file called `dbt_project.yml`. It defines the directory of the dbt project and other project configurations.
```

Edit `dbt_project.yml` to set up common project configurations such as:

<div align="center">

| YAML key  | Value description  |
| :--- | :--- |
| [name](/reference/project-configs/name) | Your project’s name in [snake case](https://en.wikipedia.org/wiki/Snake_case) |

> | [version](/reference/project-configs/version) | The [dbt Core version](/docs/dbt-versions/core) your project is using |

Reviewer 2: Although [`version` is a required parameter](https://deploy-preview-2190--docs-getdbt-com.netlify.app/reference/project-configs/version) within `dbt_project.yml`, "it is not currently meaningfully used by dbt"!

It can be _anything_ the user chooses as long as it is in `number.number.number` format. It literally won't affect anything no matter what they put.

So I'd suggest a clarification similar to the following:
```suggestion
| [version](/reference/project-configs/version) | Version of your project |
| [require-dbt-version](/reference/project-configs/require-dbt-version) | Restrict your project to only work with a range of [dbt Core versions](/docs/dbt-versions/core) |
```

Up to you if you actually want to include a row for `require-dbt-version` or not.

| [macro-paths](/reference/project-configs/macro-paths) | Directories to where your macros live |
| [snapshot-paths](/reference/project-configs/snapshot-paths) | Directories to where your snapshots live |
| [docs-paths](/reference/project-configs/docs-paths) | Directories to where your docs blocks live |
| [vars](/docs/build/project-variables) | Project variables you want to use for data compilation |

</div>

For complete details on project configurations, see [dbt_project.yml](/reference/dbt_project.yml).

## New projects

You can create new projects and [share them](/docs/collaborate/git-version-control) with other people by making them available on a hosted git repository like GitHub, GitLab, and BitBucket.

After you set up a connection with your data platform, you can [initialize your new project in dbt Cloud](/docs/develop/getting-started/getting-set-up/setting-up-bigquery#initialize-your-dbt-project) and start developing. Or, run [dbt init from the command line](/reference/commands/init) to set up your new project.

During project initialization, dbt creates sample model files in your project directory to help you start developing quickly.

## Sample projects

> If you want to explore dbt projects more in-depth, you can clone dbt Lab’s [Jaffle shop](https://github.com/dbt-labs/jaffle_shop_duckdb) on GitHub. It's a runnable project that contains sample configurations and helpful notes.

Reviewer 2: There's multiple different repos with "jaffle_shop" in the name. I've updated it below to point at the canonical one.
```suggestion
If you want to explore dbt projects more in-depth, you can clone dbt Lab’s [Jaffle shop](https://github.com/dbt-labs/jaffle_shop) on GitHub. It's a runnable project that contains sample configurations and helpful notes.
```

Author: fantastic, ty! i was totally wondering about that

> If you want to explore dbt projects more, you can clone dbt Lab’s [Jaffle shop](https://github.com/dbt-labs/jaffle_shop_duckdb) on GitHub. It's a runnable project that contains sample configurations and helpful notes.

Reviewer: ```suggestion
If you want to explore dbt projects more in-depth, you can clone dbt Lab’s [Jaffle shop](https://github.com/dbt-labs/jaffle_shop_duckdb) on GitHub. It's a runnable project that contains sample configurations and helpful notes.
```
